app/services/bolognese/writers/hyku_addons_work_form_fields_writer.rb

```diff
@@ -466,6 +466,8 @@ def hash_from_date(field, dates)
 
         # `dates` is an array of hashes containing a string date and named dateType
         def collect_date(type)
+          return unless dates.present?
```

Comment:
Just a sanity check. Did you want to check that the array is present or that it has at least one item?